pkg/identity/cache/local.go

```diff
@@ -84,8 +84,11 @@ func (l *localIdentityCache) lookupOrCreate(lbls labels.Labels, oldNID identity.
 	l.mutex.Lock()
 	defer l.mutex.Unlock()
 
-	stringRepresentation := string(lbls.SortedList())
-	if id, ok := l.identitiesByLabels[stringRepresentation]; ok {
+	repr := lbls.SortedList()
+	// Not converting to string saves an allocation, as byte key lookups into
+	// string maps are optimized by the compiler, see
+	// https://github.com/golang/go/issues/3512.
+	if id, ok := l.identitiesByLabels[string(repr)]; ok {
```

Comment:
Not directly related to the byte/string conversion, but I'm curious if the locking should be deferred until after converting labels to bytes to reduce possible contention by shrinking the critical section.

```suggestion
	repr := lbls.SortedList()
	// Not converting to string saves an allocation, as byte key lookups into
	// string maps are optimized by the compiler, see
	// https://github.com/golang/go/issues/3512.
	l.mutex.Lock()
	defer l.mutex.Unlock()
	if id, ok := l.identitiesByLabels[string(repr)]; ok {
```

Reply:
Yep, that's a good idea, thanks.